server/api.go

```diff
@@ -931,3 +942,124 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request) {
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	ctx := context.Background()
+	githubClient := p.githubConnect(*info.Token)
+
+	repositories, _, apiErr := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{})
+	if apiErr != nil {
+		mlog.Error(apiErr.Error())
+		return
+	}
+
+	resp, _ := json.Marshal(repositories)
+	w.Write(resp)
+}
+
+func (p *Plugin) createIssue(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, fmt.Sprintf("Request: %s is not allowed, must be POST", r.Method), http.StatusMethodNotAllowed)
+		return
+	}
+
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	dec := json.NewDecoder(r.Body)
+	if err := dec.Decode(&issue); err != nil {
+		mlog.Error("Error decoding JSON body", mlog.Err(err))
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Title == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue title.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Body == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue description.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Repo == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid repo name.", StatusCode: http.StatusBadRequest})
+		return
+	}
+	ctx := context.Background()
+	gh_issue := &github.IssueRequest{
+		Title: &issue.Title,
+		Body:  &issue.Body,
+	}
+
+	// Make sure user have a connected github account
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	githubClient := p.githubConnect(*info.Token)
+
+	// call Create(ctx context.Context, owner string, repo string, issue *IssueRequest)
+	result, resp, api_err := githubClient.Issues.Create(ctx, info.GitHubUsername, issue.Repo, gh_issue)
```

Comment:
Why does the issue get created under the org `info.GitHubUsername`? What happens if the issue should be created in an org repo?

Reply:
First off, good catch.  I've been thinking about this here's some of the issues.  I'd like to discuss this before moving on to implementation.

1) The repo selector currently list repos that I have available but doesn't include the Mattermost Org.  I will need to address this first.  There is a `GithubOrg` setting in the plugin settings.  I am curious if this is being set to `mattermost` in the community plugin?  I believe the implementation should by default get the repos you are a member of and if `GitHubOrg` is set, the repos should show only mattermost org repos.  

2) In the code above, you are correct and the `owner` should come from the `owner/repo` that will be specified in the create-issue modal. This could be a user repo, or an org.  My suggestion is to fix the item in 1) first.  Doing so will get us the correct `owner`/`repo` values to add to teh `githubClient.Issues.Create()` command

Reply:
3) if we use the org mode and only show `mattermost` org repos, that will reduce restrict us to create issues in personal repos.  For instance.  We have the `wrangler` channel and could not create an issue in the repo because it is under Gabe, not Mattermost

Reply:
1, Yes, `GithubOrg` is set to `mattermost` on the community server. Agreed with your conclusion.
2. :+1: 
3. While is see that there might be some rough edge cases for us, not showing "non org" repos still seams consistent to me. E.g. "non org" notifications are omitted.

Reply:
Please see https://github.com/mattermost/mattermost-plugin-github/pull/239/commits/d2a20760d413d4b4020afeff3edcce728a2dedf6 for the fix.  Basically, the repo value sent back from the client is now <owner_or_org>/<reponame>.  This is split by the server-side code to correctly determine the issue repo destination.

Reply:
1. You can test the org functionality by making your own org on GitHub and scoping the plugin setting to that org @jfrerich 

If the org is set, I think we should restrict to only creating issues in the org's repos. That's currently how notifications operate in the plugin.